packages/flutter/lib/src/widgets/router.dart

```diff
@@ -967,6 +967,10 @@ class ChildBackButtonDispatcher extends BackButtonDispatcher {
 
   @override
   void takePriority() {
+    if (parent is ChildBackButtonDispatcher) {
```

Comment:
This is for the second improvement

Reply:
Second improvement: calling take priority on a back button will immediately hook up all parent back button dispatchers so that the user do not need to call take priority on every single back button dispatcher on the ancestor chain.

This makes sense to me as a feature. Maybe it should be configurable via a parameter to this method whether you want that or not?